internal/zinc-testing/src/main/scala/xsbti/TestCallback.scala

```diff
@@ -153,6 +153,8 @@ class TestCallback extends AnalysisCallback2 {
   override def isPickleJava: Boolean = false
 
   override def getPickleJarPair = Optional.empty()
+
+  override def getSourceInfos: ReadSourceInfos = null
```

Comment:
Should it be `SourceInfos.empty` instead?

Reply:
`TestCallback.scala` lives in `zinc-testing`, therefore it only has access to `compiler-interface` (I assume the reason behind it is that `zinc-testing` should not depend on any zinc internal.

```scala
lazy val zincTesting = (projectMatrix in internalPath / "zinc-testing")
  .dependsOn(compilerInterface)
```

Since `SourceInfos.empty` lives in `zinc-core`, `TestCallback.scala` cannot directly use it.

Reply:
However, I agree that a `null` is a code smell. I had added `TestSourceInfos` class providing an `ReadSourceInfos` implementation for testing purpose.